client/client.go

```diff
@@ -455,13 +459,23 @@ func (c *client) processTSORequests(stream pdpb.PD_TsoClient, requests []*tsoReq
 	}
 
 	physical, logical := resp.GetTimestamp().GetPhysical(), resp.GetTimestamp().GetLogical()
+	failpoint.Inject("timestampDecrease", func() {
+		physical = c.lastPhysical - 1
+	})
 	// Server returns the highest ts.
 	logical -= int64(resp.GetCount() - 1)
 	c.finishTSORequest(requests, physical, logical, nil)
 	return nil
 }
 
 func (c *client) finishTSORequest(requests []*tsoRequest, physical, firstLogical int64, err error) {
+	if tsLessEqual(physical, firstLogical, c.lastPhysical, c.lastLogical) {
+		err = errors.Errorf("timestamp fallback, newly acquired ts (%d,%d) is less or equal to last one (%d, %d)",
```

Comment:
can we panic here? Returning an error will be very likely to ignored because the caller usually has some retry scheme.

Reply:
If sometimes later the new TS is greater than previous, we can go on running.

Reply:
When this happens, it is likely that a serious problem has occurred. (Even if you don't return ts here, other clients may return ts that break the consistency)

Reply:
ok